File: finger/finger.py
import sys
from cryptography import x509
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.serialization import Encoding
from cryptography.hazmat.backends import default_backend
import binascii

# Get the filename from the command line arguments
if len(sys.argv) != 2:
    print("Usage: script.py <filename>")
    sys.exit(1)

filename = sys.argv[1]

# Load the certificate
with open(filename, 'rb') as cert_file:
    cert_data = cert_file.read()
    cert = x509.load_pem_x509_certificate(cert_data, default_backend())

# Compute the SHA256 fingerprint
fingerprint = cert.fingerprint(hashes.SHA256())

# Convert to hex
fingerprint_hex = binascii.hexlify(fingerprint).decode()

# Print the lowercase fingerprint without colons
print(fingerprint_hex.lower())

File: finger/finger.go
package main

import (
	"crypto/sha256"
	"crypto/x509"
	"encoding/pem"
	"fmt"
	"io/ioutil"
	"os"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Printf("Usage: %s <filename>\n", os.Args[0])
		os.Exit(1)
	}

	filename := os.Args[1]

	// Read the certificate file
	certPEM, err := ioutil.ReadFile(filename)
	if err != nil {
		fmt.Printf("Failed to read file: %v\n", err)
		os.Exit(1)
	}

	// Parse the PEM-encoded certificate
	block, _ := pem.Decode(certPEM)
	if block == nil {
		fmt.Println("Failed to parse certificate")
		os.Exit(1)
	}

	cert, err := x509.ParseCertificate(block.Bytes)
	if err != nil {
		fmt.Printf("Failed to parse certificate: %v\n", err)
		os.Exit(1)
	}

	// Compute the SHA256 fingerprint
	fingerprint := sha256.Sum256(cert.Raw)

	// Print the lowercase hexadecimal fingerprint
	for i, byte := range fingerprint {
		if i > 0 {
			fmt.Print("")
		}
		fmt.Printf("%02x", byte)
	}
	fmt.Println()
}
